### DISCUSSION TITLE
Luhman's naming scheme and sorting notes

### DISCUSSION THREAD

**Discussion by gour** [ahyatt/ekg#86] [Q&A]
Hello!

I'm very enthusiastic about adopting `ekg` as my solely solution for note-taking...
Tried to use analog method – pen & index cards for my Zettelkasten, but found out that manually writing/linking cards is a bit tedious and time-consuming, so I believe that digital method is advantageous here. 

However, I still prefer to use pen & index cards while reading books, but plan to enter them into digital system after processing them.

One of the nice features in Luhman's method in his naming scheme – "arbitrary strings of alphanumerical characters that can be used to establish sequential relations between files at the level of their file name (e.g., 1, 1a, 1b, 1b1, 1b2, …).”  which can provide insight into the context how one has arrived to certain point(s).

In `ekg` I can use such tags ("1", "1a"...), to mimic Luhman's naming scheme, but could not find a way to sort notes where natural sorting order would list notes showing the above-mentioned context?

**Comment by ahyatt** [comment]
I think such a thing is possible, and not even very hard.  At least, I think so, I've never tried to use this sort of scheme.  A few things to note:

1.  It probably is the case that a numbering scheme should be a new property.  Unless you want to refer to it in other notes, in which case it does make sense as a tag.  But if it's a tag, then you can't distinguish between a note's number and a number you are just using for tagging purposes, as a way to link notes.  So probably it should be a new property.
2. Right now we sort by modification date (newest first) in most notes views.  It should be relatively easy to make the sorting method a variable which can be configured, in which case you can substitute your Luhman tag sorting.  If you'd like to contribute such a change, please send a pull request.  Otherwise, it's a good idea, but I don't have immediate plans to implement it.  There's a few other things I plan to work on first.

**Reply by ahyatt** [reply]
Now that I think of this more, the Luhman numbers might be best added in an additional module, which should theoretically possible (and if it isn't possible, we should make it possible).

**Reply by gour** [reply]
> Now that I think of this more, the Luhman numbers might be best added in an additional module, which should theoretically possible (and if it isn't possible, we should make it possible).

I'm glad to hear it is possible. Moreover, I read that Ekg's source is approachable, but for noob as myself it will certainly take some time before being able to PR.

**Comment by gour** [comment]
> I think such a thing is possible, and not even very hard. 

:smiley: 

>     1. It probably is the case that a numbering scheme should be a new property.  Unless you want to refer to it in other notes, in which case it does make sense as a tag.  But if it's a tag, then you can't distinguish between a note's number and a number you are just using for tagging purposes, as a way to link notes.  So probably it should be a new property.

How it would be entered in such case?

>     2. Right now we sort by modification date (newest first) in most notes views.  It should be relatively easy to make the sorting method a variable which can be configured, in which case you can substitute your Luhman tag sorting.  If you'd like to contribute such a change, please send a pull request.  

Unfortunately, I do not speak Elisp (yet)…but might try to venture into it…

> Otherwise, it's a good idea, but I don't have immediate plans to implement it.  There's a few other things I plan to work on first.

Sure. First things first. :wink:

@jayrajput Does your `ekg-denote` module handle Denote's signatures?

I wonder if I continue with Denote, whether it would be possible to easily migrate to Ekg in case that this "Lluhman module” gets ever implemented?

**Comment by jayrajput** [comment]
I will need to look more into the denote signature and support for in `ekg-denote`. I will share an update when possible. Things are busy on my end.  Btw, I am playing with paper and pen as well to manage notes. There are fews things which are just nice with paper and pen. I am trying the bullet journal technique.

**Comment by gour** [comment]
> I will need to look more into the denote signature and support for in `ekg-denote`. I will share an update when possible.

Thank you!

> Btw, I am playing with paper and pen as well to manage notes. There are fews things which are just nice with paper and pen. 

Yes, pen & paper have some merits.

> I am trying the bullet journal technique.

For events/tasks I use [Etm](https://github.com/dagraham/etm-dgraham)...

**Comment by jayrajput** [comment]
@gour Provide guidance on how do you want the signature to exist in the Ekg system. Is it OK for them to exist as tags?

I am thinking of keep them as a tag called denote-signature and then provide ability to sort them in the ekg view.

**Comment by gour** [comment]
> @gour Provide guidance on how do you want the signature to exist in the Ekg system. Is it OK for them to exist as tags?
> 
> I am thinking of keep them as a tag called denote-signature and then provide ability to sort them in the ekg view.



> @gour Provide guidance on how do you want the signature to exist in the Ekg system. Is it OK for them to exist as tags?

Exactl! :wink: 

> I am thinking of keep them as a tag called denote-signature and then provide ability to sort them in the ekg view.

It sounds good. Thank you!


> Provide guidance on how do you want the signature to exist in the Ekg system. Is it OK for them to exist as tags?
> 
> I am thinking of keep them as a tag called denote-signature and then provide ability to sort them in the ekg view.
